sirepo/job_supervisor.py

```diff
@@ -79,6 +79,60 @@ class Awaited(Exception):
     pass
 
 
+class ServerReq(PKDict):
+
+    def copy_content(self):
+        return copy.deepcopy(self.content)
+
+    def pkdebug_str(self):
+        c = self.get('content')
+        if not c:
+            return 'ServerReq(<no content>)'
+        return pkdformat('ServerReq({}, {})', c.api, c.get('computeJid'))
+
+    async def receive(self):
+        s = self.content.pkdel('serverSecret')
+        # no longer contains secret so ok to log
+        assert s, \
+            'no secret in message: {}'.format(self.content)
+        assert s == sirepo.job.cfg.server_secret, \
+            'server_secret did not match'.format(self.content)
+        self.handler.write(await _ComputeJob.receive(self))
+
+
+class SlotProxy(PKDict):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs, _value=None)
```

Comment:
**kwargs must be the last thing with no trailing comma in py2. I guess it is valid in py3. Still looks kind of odd.

Reply:
374a8cb5